### DISCUSSION TITLE
Speed up startup on rpi zero w

### DISCUSSION THREAD

**Discussion by arne123** [MiczFlor/RPi-Jukebox-RFID#1274] [General]
Dear all,

one general issue I see  is the long boot time with the phoniebox.
The boot time makes it a little bit annoying especially for young children or impatient persons like me to use it.
I think this is also one of the major differences to commercial products from usability point of view.

So I,d like to share some improvements I made so far, collect some feedback (Maybe some things are having a negative side effect I am not aware of), and of course collect further ideas for more boot speed.

The following steps I did with at least some success:

- Starting Network stuff after audio stuff (I want music in first place network later)
- upgrading mpd to the new version  (0.23 here, the raspbian one is way older) 
- mpd  disabled auto-update
- avoid parallel starting of other services in parallel to mpd
- shortened the startup sound (trimmed leading and trailing empty parts)

At the moment I have audio available 32s after the boot.
The bootchart looks looks like this:
https://gist.githubusercontent.com/arne123/25068300a6bc18a189103a1f13b2cd53/raw/f327f88df4df0e04a806e6db6af94ed2ea6098f4/boot.svg

what I still want to do:
- searching for more services which are not required and can be disabled or postponed after audio
- skip systemd-fsck@dev, or execute it during shutdown (1.7s)
- see if there are some possibilities to shorten the systemd start itself (~3s)
- see if the kernel period can be shortened (3s)
- see if dev-mmcblk0p2 can be shortened (7s)
   - skip systemd-fsck-root service, or execute it during shutdown (1.3s) 
- see if mpd startup can be shortened more
- check if the startup script can be modified in some ways, e.g. playing sound earlier (in parallel to mpd)(~3s)

So my goal is to boot til audio in at least 20s.

Question to the community:
  SD-card , I thought I bought a good sd card(80MB/s), but I doubt this somehow now.
  Is anybody also using an rpi zero w and can compare the boot times for some services?
  Is there some potential in the sd card?

br
  Arne

**Comment by s-martin** [comment]
Awesome, thanks for looking into that. That's s problem, which affects many users.

We already have a issue for that #1110, could you add your results and continue the discussion there?

Thanks 👍
